lib/battle/BattleHex.h

```diff
@@ -83,29 +91,31 @@ struct DLL_LINKAGE BattleHex //TODO: decide if this should be changed to class f
 	BattleHex cloneInDirection(EDir dir, bool hasToBeValid = true) const;
 	BattleHex operator+(EDir dir) const;
 
-	/// returns all valid neighbouring tiles
-	std::vector<BattleHex> neighbouringTiles() const;
```

Comment:
Already mentioned this, but I think this method should be restored and it should return cached / precomputed values - more easy to read code than what you have now